### ISSUE TITLE
[4.0.0-beta.17] Conditional ion-select-option not updating

### ISSUE THREAD

**Issue by RHinderiks** [ionic-team/ionic#16453]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```
Ionic:

   ionic (Ionic CLI)             : 4.3.1 (/Users/rh/.nvm/versions/node/v8.11.4/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.0.0-beta.16
   @angular-devkit/build-angular : 0.10.5
   @angular-devkit/schematics    : 7.0.5
   @angular/cli                  : 7.0.5
   @ionic/angular-toolkit        : not installed

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : android 7.1.2, ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.2, cordova-plugin-ionic-webview 2.1.0, (and 13 other plugins)

System:

   Android SDK Tools : 26.1.1 (/Users/rh/Library/Android/sdk)
   ios-deploy        : 1.9.2
   ios-sim           : 6.0.0
   NodeJS            : v8.11.4 (/Users/rh/.nvm/versions/node/v8.11.4/bin/node)
   npm               : 6.4.1
   OS                : macOS High Sierra
   Xcode             : Xcode 10.1 Build version 10B61
```

**Describe the Bug**
I have a select box in which the items contain a integer value. When this integer value is decreased below zero we try to hide the item with an *ngIf. However the item remains and the label of the ion-select does not update. This only happens when the limit of 0 is reached and the ngif is triggered. 

if is add a button that ngif's the entire ion-select and toggle it on and off then it does update.

**Steps to Reproduce**
1. Add several options to a ion-select
2. Add *ngIf to the options
3. Dynamicly change the value that your ngiffing to trigger de ngif with a button or something
4. open the selectbox
5. be amazed the value is still there

**Related Code**
` <ng-container *ngFor="let productCredit of productCredits">
            <ion-select-option *ngIf="productCredit.bookableCredit > 0" [value]="productCredit">{{ productCredit.label }}</ion-select-option>
        </ng-container>
<ion-button (tap)="changedBookableCreditToLessThanZero()">Push me<ion-button>`

**Expected Behavior**
The values in a ion-select should update visually, in my case the should dissappear because i ng-iffed them.

**Additional Context**
List any other information that is relevant to your issue. Stack traces, related issues, suggestions on how to fix, Stack Overflow links, forum links, screenshots, OS if applicable, etc.

**Comment by d0x** [comment]
Could be related to #15885.

**Comment by RHinderiks** [comment]
Could be but it does not look like that ticket has been picked up.

**Comment by RHinderiks** [comment]
This is still an major issue in beta.17!

**Comment by bryce13950** [comment]
> Could be related to #15716.

I think you are correct. Thanks for connecting them. I think once this is fixed both should be addressed.

**Comment by RanaProgrammer** [comment]
is there any chance we can handle the dynamic selection of options via `value` on ion-select. anyone tried that?

i am facing an issue in rendering the selected options. they become visible when i click the select component. i think there is some issue with change detection.

**Comment by liamdebeasi** [comment]
Hi there,

Thanks for the issue! We have pushed a nightly build containing some fixes for loading/re-loading select options. Would you be able to try it out in your app and let me know if the issue you are running into here is fixed? (`npm i @ionic/core@dev @ionic/angular@dev`)

Thanks for your patience as we work to resolve this issue!

**Comment by Maarcel** [comment]
Hi @liamdebeasi, unfortunately it still does not work..

**Comment by liamdebeasi** [comment]
Hi @Maarcel,

Thanks for checking! Do you have a code snippet I can use to reproduce this issue?

Thanks!

**Comment by Maarcel** [comment]
```
<ion-item>
    <ion-label>Numbers</ion-label>
    <ion-select placeholder="Choose">
        <ion-select-option *ngFor="let number of numbers">{{ number }}</ion-select-option>
    </ion-select>
</ion-item>
```

For example i have an array with numbers. When i change the numbers array this has no effect to the `ion-select-options`

**Comment by ShaneJohnsonCC** [comment]
Hey @liamdebeasi !

Is there an update on this? I'm the same exact scenario as @Maarcel, however my select options will update only if the array of numbers is greater than the previous.

So if it started at 25 and I updated it to 30, it would work. But if I then change it to 29 or lower, no update occurs...

**Comment by liamdebeasi** [comment]
Hi everyone,

There are some changes to Stencil actively being worked on that we think may resolve this issue. I am going to do some testing with it later this week and will post here when I have an update.

Also linking https://github.com/ionic-team/ionic/issues/17344 since it seems to be related.

Thanks!

**Comment by Majed93** [comment]
I'm seeing a similar issue with ion-item-options. If i add *ngIf - i can't slide the ion-item-sliding

**Comment by liamdebeasi** [comment]
Hi everyone,

The updates to Stencil need a little bit more time. I will update everyone when I more to share.

Thanks!

**Comment by ShaneJohnsonCC** [comment]
Hey @liamdebeasi !

Do you have an estimate timeline on this? Or maybe a temporary fix we can do on our end?

Thanks!

**Comment by Ladeiras** [comment]
Hi,
Anyone with an elegant workaround for this?
Thank you :)

**Comment by ShaneJohnsonCC** [comment]
Ditto what @Ladeiras asked. Just to be persistent :) A part of my app **really** depends on this working properly

**Comment by kamalessence** [comment]
+1

**Comment by BOGDANOLEKSANDR** [comment]
**Have the problem with `selected` option of <ion-select-option>**

Ionic:
```
   ionic (Ionic CLI)             : 4.12.0 (C:\Users\admin\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework               : @ionic/angular 4.5.0-dev.201905231504.7ab9479
   @angular-devkit/build-angular : 0.13.8
   @angular-devkit/schematics    : 7.2.4
   @angular/cli                  : 7.3.8
   @ionic/angular-toolkit        : 1.4.1

Cordova:

   cordova (Cordova CLI) : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms     : android 8.0.0, browser 6.0.0
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 3.1.2, (and 7 other plugins)

System:

   NodeJS : v10.15.3 (C:\Program Files\nodejs\node.exe)
   npm    : 6.4.1
   OS     : Windows 10```

HTML:
```
<ion-select multiple="true" cancelText="{{ 'no' | translate }}" okText="{{ 'yes' | translate }}"
				class="select-center-wide" placeholder="Select Location" (ionChange)="onLocationChange($event)">
				<ion-select-option *ngFor="let location of locations" selected="{{location.selected}}" value="{{location.location_id}}">
					{{location.name}}
				</ion-select-option>
			</ion-select>
```

Data:
```Array(9)
0: {location_id: "1", name: "Kallio", selected: true}
1: {location_id: "2", name: "Mäkelänrinne", selected: true}
2: {location_id: "3", name: "Kontula", selected: false}
3: {location_id: "4", name: "Töölö", selected: false}
4: {location_id: "5", name: "Malmi", selected: false}
5: {location_id: "6", name: "Siltamäki", selected: false}
6: {location_id: "7", name: "Kallio Toimisto", selected: false}
7: {location_id: "8", name: "Pasila", selected: false}
8: {location_id: "9", name: "Vuosaari", selected: false}
```

It really ignores the preselection.

**Comment by BOGDANOLEKSANDR** [comment]
And this `https://github.com/ionic-team/ionic/issues/17225` doesn't help

**Comment by BDominik** [comment]
Any solution on this?
I already spent more then 3 hours searching for fixes, and nothing...

**Comment by david0178418** [comment]
As a workaround, I'm just rendering two distinct select boxes that I'm toggling between:

```html
<ion-select *ngIf="myCondition">
	<ion-select-option>A</ion-select-option>
	<ion-select-option>B</ion-select-option>
	<ion-select-option>C</ion-select-option>
</ion-select>
<ion-select *ngIf="!myCondition">
	<ion-select-option>A</ion-select-option>
	<ion-select-option>C</ion-select-option>
</ion-select>
```

It's brute force, but it works.

**Comment by BDominik** [comment]
Thank you for your answer I did exactly the same thing. :D
Not nice but at least working.

**Comment by eRendon** [comment]
other solution on this?
the solution at @david0178418 is not posible to my. need any solution because selects options are dynamics every time

**Comment by liamdebeasi** [comment]
A fix for this is available in Ionic 4.9.0. For any new bugs, please create a new issue.

Thanks!
